src/Error/Renderer/WebExceptionRenderer.php

```diff
@@ -411,12 +412,29 @@ protected function getHttpCode(Throwable $exception): int
      * Generate the response using the controller object.
      *
      * @param string $template The template to render.
+     * @param bool $skipControllerCheck Skip checking controller for existence of
+     *   method matching the exception name.
      * @return \Cake\Http\Response A response object that can be sent.
      */
-    protected function _outputMessage(string $template): Response
+    protected function _outputMessage(string $template, bool $skipControllerCheck = false): Response
     {
         try {
-            $this->controller->render($template);
+            $method = $this->method ?: $this->_method($this->error);
+
+            if (!$skipControllerCheck && method_exists($this->controller, $method)) {
+                $this->controller->viewBuilder()->setTemplate($method);
+
+                $reflectionMethod = new ReflectionMethod($this->controller, $method);
+                $result = $reflectionMethod->invoke($this->controller);
```

Comment:
Should we deprecate the exception renderer custom methods? Perhaps we wait a release or two?

Reply:
I would wait a while.